edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/brokerConnection/MqttBrokerConnector.cs

```diff
@@ -369,9 +404,19 @@ static async Task SubscribeAsync(MqttClient client, IReadOnlyCollection<IMessage
                             Enumerable.Range(1, subscriber.Subscriptions.Count).Select(i => (byte)1).ToArray());
                     }
 
-                    await acksArrived.WaitAsync();
-
-                    client.MqttMsgSubscribed -= ConfirmSubscribe;
+                    try
```

Comment:
For some reason I can’t add comments to code that was not changed in this PR: allqosgranted should be AtomicBoolean and expectedAckCount could use AtomicLong